servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -9984,16 +10017,35 @@ public void run() {
     }
 
     public void recordMetrics(ResourceContext ctx, int httpStatus) {
-        final String principalDomainName = getPrincipalDomain(ctx);
-        final String domainName = getRequestDomainName(ctx);
-        final Object timerMetric = getTimerMetric(ctx);
-        final String httpMethod = (ctx != null) ? ctx.getHttpMethod() : null;
-        final String apiName = (ctx != null) ? ctx.getApiName() : null;
-        final String timerName = (apiName != null) ? apiName + "_timing" : null;
-        metric.increment("zms_api", domainName, principalDomainName, httpMethod, httpStatus, apiName);
-        metric.stopTiming(timerMetric, domainName, principalDomainName, httpMethod, httpStatus, timerName);
+        try {
+            final String principalDomainName = getPrincipalDomain(ctx);
+            final String domainName = getRequestDomainName(ctx);
+            final Object timerMetric = getTimerMetric(ctx);
+            final String httpMethod = (ctx != null) ? ctx.getHttpMethod() : null;
+            final String apiName = (ctx != null) ? ctx.getApiName() : null;
+            final String timerName = (apiName != null) ? apiName + "_timing" : null;
+            metric.increment("zms_api", domainName, principalDomainName, httpMethod, httpStatus, apiName);
+            metric.stopTiming(timerMetric, domainName, principalDomainName, httpMethod, httpStatus, timerName);
+        } catch (Exception e) {
+            LOG.warn("Got exception during recordMetrics: {}", e.getMessage(), e);
```

Comment:
let's make it as error so they're logged in production otherwise we'll never know that something is broken